### ISSUE TITLE
Unexpected "importSourceResolveFailure"

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#4653]
## Environment data

-   Language Server version: v2023.7.41 PreRelease
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.11.4 Anaconda

## Code Snippet

t.pyi
```python
pass
```

test.py
```python
from . import t  # pyright: ignore[reportMissingModuleSource, reportUnusedImport]
```

## Repro Steps

1. Create a ".pyi" file.
2. Create a ".py" file and try to import from the ".pyi" file in previous step.
3. Suppress "reportMissingModuleSource" and maybe "reportUnusedImport".

## Expected behavior

No error should be reported.

## Actual behavior

Receives an error named "importSourceResolveFailure".

## Logs
Nothing interesting.

## Additional Info

The code passes type-checking with Pyright 1.1.319.

**Comment by rchiodo** [comment]
This issue has the same root cause as this one: https://github.com/microsoft/pylance-release/issues/4651

Should be fixed shortly.

**Comment by rchiodo** [comment]
2023.7.40 doesn't have this issue as it's a recent regression.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.7.42, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023742-28-July-2023-PreRelease)
